engines/bops_config/app/controllers/bops_config/application_controller.rb

```diff
@@ -22,5 +22,15 @@ def set_back_path
       session[:back_path] = request.referer if request.get?
       @back_path = session[:back_path]
     end
+
+    def application_type_id
+      if request.path_parameters.key?(:application_type_id)
+        Integer(params[:application_type_id])
+      else
+        Integer(params[:id])
+      end
+    rescue
+      raise ActionController::BadRequest, "Invalid application type id: #{params[:id]}"
```

Comment:
In the exception we always reference `id` but in the body we sometimes reference `application_type_id`. Is this going to matter, if the exception is thrown in the first branch of the conditional, or would `id` still be meaningful here?

Reply:
```ruby
    def application_type_id
      param_value = params[request.path_parameters.key?(:application_type_id) ? :application_type_id : :id]
      Integer(param_value)
    rescue
      raise ActionController::BadRequest, "Invalid application type id: #{param_value}"
    end
```